### PR TITLE
dockerize integration tests

### PR DESCRIPTION
**Pull request by xvello**
### What does this PR do?

As our integration tests rely on access to docker (either to run application containers, or even to test docker AD & metrics collection), this allows running them in [a remote sandboxed docker environment](https://circleci.com/docs/2.0/building-docker-images/) with the `--remote-docker` flag. This will allow to run them in CCI, but also on local `docker-machines`.

The go test is compiled, packaged in a docker image and run there. The dockerization process is transparent to the `go test` command, which means it's impact on invoke is pretty small.

This PR is also reworks the dsd origin detection to use `docker-compose`. Manually launching containers with the golang client is tedious, while compose handles image building, running containers and cleanup very reliably.

I also fixed the etcd and zk integration tests, but could not fix the dockerlistener test: it only works when no container is running on start. We'll need to rework it in another PR. It's disabled for now.

### Testing

- It does fail if a test fails: https://circleci.com/gh/DataDog/datadog-agent/5122
- It does work if all tests OK: https://circleci.com/gh/DataDog/datadog-agent/5129

### CODE REVIEW

File: tasks/docker.py [hunk 1]
```diff
@@ -21,3 +27,70 @@ def integration_tests(ctx, skip_image_build=False, skip_build=False):
...
+    Run a go test in a remote docker environment and pipe its output to stdout
+    """
+    import docker
+
+    client = docker.from_env()
+    temp_folder = tempfile.mkdtemp(prefix="ddtest-")
+
+    ctx.run("cp %s %s/test.bin" % (binary, temp_folder))
+
+    with open("%s/Dockerfile" % temp_folder, 'w') as stream:
```
Comment (mfpierre) [inline tasks/docker.py]: This part is a little sketchy, I'm wondering if we should not have a real DockerFile in `test/integration` ?

File: tasks/docker.py [hunk 2]
```diff
@@ -21,3 +27,70 @@ def integration_tests(ctx, skip_image_build=False, skip_build=False):
     print("Starting docker integration tests")
     env = {"DOCKER_IMAGE": DOGSTATSD_TAG}
     ctx.run("./test/integration/docker/dsd_alpine_listening.sh", env=env)
+
+
+@task
+def dockerize_test(ctx, binary, skip_cleanup=False):
+    """
+    Run a go test in a remote docker environment and pipe its output to stdout
```
Comment (mfpierre) [inline tasks/docker.py]: You should probably tell that's it'll not work on all platform

**Review by mfpierre — CHANGES_REQUESTED** [review]
Looks good overall, thanks for fixing some tests 👍 

If you can just add a comment that the dockerized test is platform specific, the other comment is optional

File: tasks/docker.py (hunk 1 above)
Comment (xvello) [reply tasks/docker.py]: As dockerfiles don't support conditionnal layers, I don't see how to optionnaly add the testdata folder if it exists. We'd need to open the dockerfile in python to add the file anyway.
But I aggree with you, that's not great for readability, I'll investigate how to improve on that.

**Review by mfpierre — APPROVED** [review]

**Pull request merged**